ci_framework/roles/env_op_images/tasks/main.yml

```diff
@@ -0,0 +1,104 @@
+    PATH: "{{ cifmw_path | default(ansible_env.PATH) }}"
+  register: oc_installed
+  failed_when: false
+
+- name: Check if OpenStackControlPlane is setup
+  when:
+    - oc_installed is defined
+    - oc_installed.rc == 0
+    - cifmw_openshift_kubeconfig is defined
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  ansible.builtin.command:
+    cmd: oc get OpenStackControlPlane -o=jsonpath='{.items[*].status.conditions[?(@.type=="Ready")].status}'
+  register: openstack_installed
+  failed_when: false
+
+- name: Get openstack operator images
+  when:
+    - openstack_installed.stdout == "True"
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  block:
+    - name: Get images from the csv
+      ansible.builtin.shell: |
+        csv_name=$(oc get csv -n {{ cifmw_install_yamls_defaults['OPERATOR_NAMESPACE'] }} -l operators.coreos.com/openstack-operator.openstack-operators -o=jsonpath='{.items[*].metadata.name}')
+        oc get csv -n {{ cifmw_install_yamls_defaults['OPERATOR_NAMESPACE'] }} ${csv_name} -o=jsonpath='{.spec.install.spec.deployments[*].spec.template.spec.containers[?(@.name=="manager")].env}'
+      register: csv_images
+
+    - name: Extract env variable name and images
+      loop: "{{ csv_images.stdout| regex_findall('\\{[^}]+\\}') }}"
+      ansible.builtin.set_fact:
+        cifmw_openstack_service_images_content: "{{ cifmw_openstack_service_images_content | default({}) | combine({(item | from_json).name: (item | from_json).value}) }}"
+
+    - name: Get all the pods in openstack-operator namespace
+      kubernetes.core.k8s_info:
+        kind: Pod
+        namespace: "{{ cifmw_install_yamls_defaults['OPERATOR_NAMESPACE'] }}"
+        kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+        api_key: "{{ cifmw_openshift_token | default(omit)}}"
+        context: "{{ cifmw_openshift_context | default(omit)}}"
+        field_selectors:
+          - status.phase=Running
+      register: pod_list
+
+    - name: Retrieve openstack-operator-index pod
+      vars:
+        selected_pod: "{{ pod_list.resources| selectattr('metadata.generateName', 'equalto', 'openstack-operator-index-') | list|first}}"
+      ansible.builtin.set_fact:
+        cifmw_install_yamls_vars_content:
+          OPENSTACK_IMG: "{{ selected_pod.status.containerStatuses[0].imageID }}"
+
+    - name: Get operator images and pods
+      vars:
+        selected_pod: "{{ pod_list.resources | selectattr('metadata.generateName', 'contains', 'rabbitmq-cluster-operator-') | list | first }}"
+      ansible.builtin.set_fact:
+        cifmw_openstack_operator_images_content:
+          RABBITMQ_OP_IMG: "{{ selected_pod.status.containerStatuses[0].imageID }}"
+        selected_pods: "{{ pod_list.resources | rejectattr('metadata.generateName', 'contains', 'openstack-operator-index-') | rejectattr('metadata.generateName', 'contains', 'rabbitmq-cluster-operator-') }}"
```

Comment:
Just a style comment, is good, but for consistency, we try to break long jinja2 strings in multiple lines to improve readability.